src/test/spec/sessions/mod.rs

```diff
@@ -0,0 +1,287 @@
+            client
+                .database("admin")
+                .run_command(doc, None)
+                .await
+                .unwrap();
+        }
+
+        let options = match test.session_options {
+            Some(ref options) => options.get("session0").cloned(),
+            None => None,
+        };
+        let mut session0 = Some(client.start_session(options).await.unwrap());
+        let session0_lsid = session0.as_ref().unwrap().id().clone();
+
+        let options = match test.session_options {
+            Some(ref options) => options.get("session1").cloned(),
+            None => None,
+        };
+        let mut session1 = Some(client.start_session(options).await.unwrap());
+        let session1_lsid = session1.as_ref().unwrap().id().clone();
+
+        for operation in test.operations {
+            let db = client.database(&test_file.database_name);
+            let coll = db.collection(&test_file.collection_name);
+
+            let session = match operation.session.as_deref() {
+                Some("session0") => session0.as_mut(),
+                Some("session1") => session1.as_mut(),
+                Some(other) => panic!("unknown session name: {}", other),
+                None => None,
+            };
+
+            let result = match operation.object {
+                OperationObject::Collection => {
+                    let result = operation.execute_on_collection(&coll, session).await;
+                    if test.description == "Server supports implicit sessions" {
+                        RUNTIME.delay_for(Duration::from_secs(1)).await;
+                    }
+                    result
+                }
+                OperationObject::Database => operation.execute_on_database(&db, session).await,
+                OperationObject::Session0 => {
+                    if operation.name == "endSession" {
+                        let session = session0.take();
+                        drop(session);
+                        RUNTIME.delay_for(Duration::from_secs(1)).await;
+                    } else {
+                        operation
+                            .execute_on_session(session0.as_ref().unwrap())
+                            .await;
+                    }
+                    continue;
+                }
+                OperationObject::Session1 => {
+                    if operation.name == "endSession" {
+                        let session = session1.take();
+                        drop(session);
+                        RUNTIME.delay_for(Duration::from_secs(1)).await;
+                    } else {
+                        operation
+                            .execute_on_session(session1.as_ref().unwrap())
+                            .await;
+                    }
+                    continue;
+                }
+                OperationObject::TestRunner => {
+                    match operation.name.as_str() {
+                        "assertDifferentLsidOnLastTwoCommands" => {
```

Comment:
All of these operations take in different arguments so it didn't make much sense to integrate them into the usual operation parsing -- I'll abstract this out a little more if it gets too unwieldy when the additional transactions operations are introduced.